signer/nostr-browser-signer-proxy/src/lib.rs

```diff
@@ -169,7 +170,19 @@ pub struct BrowserSignerProxy {
     shutdown: Arc<Notify>,
 }
 
-// TODO: use atomic-destructor to automatically shutdown this when all instances are dropped
+impl Drop for BrowserSignerProxy {
```

Comment:
The Drop mustn't implemented otherwise it will shutdown on the first drop. The atomic-destructor keep count of the clones and the drops and executes the `on_destroy` method when the last instance is dropped.